src/widgets/Widget.tsx

```diff
@@ -4,30 +4,79 @@ import {
   pickExternalProps,
   WidgetWrapper,
 } from '../components/WidgetWrapper';
-import { AsyncQueryEncoder } from '../helpers/query-encoder';
+import { AsyncQueryEncoder, AsyncZodQueryEncoder, MergedAsyncQueryEncoder } from '../helpers/query-encoder';
+import * as zod from 'zod';
+
+const SharedWidgetPropsSchema = zod.object({
+  originalPageUrl: zod.string().optional(),
+});
+
+type SharedWidgetProps = zod.infer<typeof SharedWidgetPropsSchema>;
+
+class SharedWidgetPropsEncoder implements AsyncQueryEncoder<SharedWidgetProps> {
```

Comment:
This wrapper class handles compatibility with our old URLs, which don't have a `sharedWidgetJson` query parameter. We can remove this class if/once it's ok to break compatibility with URLs generated before this PR was merged. @chaoran-chen When should we remove this compatibility class?

Reply:
You can remove it now :)